src/lib.rs

```diff
@@ -439,6 +450,98 @@ impl DstLayout {
         }
     }
 
+    /// Extends `self` with the layout of `field`, adding whatever padding is
+    /// needed to satisfy the field's natural alignment or `max_align`
+    /// (whichever comes first). If `self` corresponds to a type marked with
+    /// `repr(packed(N))`, then `max_align` should be `Some(N)`, otherwise
+    /// `None`.
+    ///
+    /// This method does **not** add trailing padding needed to satisfy the
+    /// type's overall alignment requirements.
+    ///
+    ///
+    #[allow(dead_code)]
+    pub(crate) const fn extend(
+        self,
+        field: DstLayout,
+        packed_modifier: Option<NonZeroUsize>,
+    ) -> Self {
+        use util::{core_layout::_padding_needed_for, max, min};
+
+        // If `max_align` is `None`, there are no alignment constraints, and the
+        // value can be defaulted to `MAX_ALIGN`.
+        let max_align = match packed_modifier {
+            Some(max_align) => max_align,
+            None => Self::MAX_ALIGN,
+        };
+
+        debug_assert!(max_align.is_power_of_two());
+        debug_assert!(self._align.get() <= max_align.get());
```

Comment:
Also assert that `field._align.get() <= Self::MAX_ALIGN`?

Reply:
I didn't mean to assert `debug_assert!(self._align.get() <= max_align.get());`, and revised the PR accordingly. I don't think we need to peek into `self` and `field` and make sure they're valid — that's the job of `DstLayout`'s constructors.

Reply:
We don't currently document anywhere that `DstLayout`'s `align` field is guaranteed to have a particular maximum value - that's just a consequence of Rust's type rules. IMO we should either explicitly document that as an invariant and add comments at construction sites proving that it's upheld, or we should assert it here out of an abundance of caution. As this code is currently written, it could break in the future if Rust raises the maximum alignment.

Reply:
I've made three changes:
1. I've verified, with Kani, that this method is robust to future increases of Rust's maximum-allowed alignment.
2. I've added debug assertions that fire if we see an alignment exceeding Rust's current maximum-allowed alignment.
3. I've replaced the `+` operations with explicit checked adds, and panic messages explaining what went wrong.